### PR TITLE
New CMake based installer.

### PR DESCRIPTION
**Pull request by oierlauzi**
Complete re-write from scratch of the installer, using CMake instead of SCons.

### CODE REVIEW

**Comment by oierlauzi** [comment]
Sonarcloud issues are from already existing code. In addition most of them request using C++20 features, which we would like to avoid for compatibility reasons.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#  All comments concerning this program package may be sent to the
+#  e-mail address 'xmipp@cnb.csic.es'
+# ***************************************************************************
+
+cmake_minimum_required(VERSION 3.16)
+
+# Define the project
+project(
+  XmippCore
+	VERSION 3.23.11
```
Comment (Ratolon) [inline CMakeLists.txt:28]: Is this the version?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,12 @@
-## devel
+## Release 3.XX.YY - TBD
```
Comment (Ratolon) [inline CHANGELOG.md:1]: Not updated with 3.23 or 3.24 versions lol

**Review by Ratolon — CHANGES_REQUESTED** [review]
Some issues with version namings but other than that OK

File: CHANGELOG.md (hunk 2 above)
Comment (oierlauzi) [reply CHANGELOG.md:1]: This was in CMakeFiles.txt

**Review by Ratolon — APPROVED** [review]
Looks good to me. Compiles in CentOS, Ubuntu and MacOS14.5

**Pull request merged**